packages/vscode-dependencies-validation/src/vscodeTypes.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  ExtensionContext,
+  OutputChannel,
+  Uri,
+  window,
+  workspace,
+  languages,
+  commands,
+  CodeActionKind,
+  DiagnosticCollection,
+} from "vscode";
+
+export type VscodeOutputChannel = Pick<
+  OutputChannel,
+  "append" | "show" | "appendLine"
+>;
+
+export type VscodeWorkspace = typeof workspace;
```

Comment:
using such "wide" types that include everything in some VSCode property
can result in more complex tests / mocks

Reply:
I use at least 6-7 methods on this type and there are no problems with tests